### ISSUE TITLE
Visitor Log: Getting "Segment not applied" notification although segment is applied.

### ISSUE THREAD

**Issue by tsteur** [piwik/piwik#7892]
See screenshot.

I'm getting this notification although the segment is actually applied to the visitor log. So 2 possibilities:

* The segment should not have been applied and the notification is correct
* The segment is applied correctly and the notification should not be shown

![segment_not_applied](https://cloud.githubusercontent.com/assets/273120/7599740/c583268a-f958-11e4-8e55-d497299c8ab9.png)

Some of my config entries include
```
[General]
browser_archiving_disabled_enforce = 1
enable_browser_archiving_triggering = 0
```

**Comment by mattab** [comment]
Good find. The popover should be displayed only for archived reports. The segment works well for Visitor Log, Visitor Profile, real time map, etc.

**Comment by tsteur** [comment]
I have actually no idea how to fix this, at least at this point of time. It might be easier after Piwik 3.0.0 and after #7822 is resolved.

The SegmentEditor is loaded in `CoreHome.index` before any report is loaded. In the initial page load an HTML attribute /JS property `isSegmentNotAppliedBecauseBrowserArchivingIsDisabled` is set that triggers the modal confirm notification. The actual report is loaded later in another XHR and the SegmentSelector does actually not know anything about the report that is loaded later. 

To solve this we'd maybe need to define for each report whether it depends on "real time logs" or "archived reports". Then, when loading a page, we'd need to check whether there is at least one report that uses "archives" and whether the config for segment is actually disabled. Eg there could be a dashboard showing only "real time log" reports, then we would not be allowed to show it. 

As the dialog is shown only on the initial page load, we would not be allowed to show the dialog if there are only "real time reports" shown, but as soon as one changes the page to another report (it is not a complete page reload, it is loaded via XHR), we'd need to show the dialog if there's an "archived report" shown. 

All this is a lot of work and probably requires a lot of refactorings. I'd rather postpone it to somewhen far after Piwik 3.0.0. Having to define whether it is a "real time report" or "archive report" is rather annoying and maybe not really doable by a 3rd party developer so our framework would ideally detect this automatically after #7470 is done

**Comment by tsteur** [comment]
@mattab are you eg to move it to mid term?

**Comment by mattab** [comment]
@tsteur thanks for analysis. Since it's not do-able in reasonable, and it is not critical in terms of user impact, moving to `Mid term`
